pkg/gui/presentation/graph/graph.go

```diff
@@ -336,17 +328,9 @@ func renderPipeSet(
 
 	// so we have our commit pos again, now it's time to build the cells.
 	// we'll handle the one that's sourced from our selected commit last so that it can override the other cells.
-	selectedPipes := []*Pipe{}
-	// pre-allocating this one because most of the time we'll only have non-selected pipes
-	nonSelectedPipes := make([]*Pipe, 0, len(pipes))
-
-	for _, pipe := range pipes {
-		if highlight && equalHashes(pipe.fromSha, selectedCommitSha) {
-			selectedPipes = append(selectedPipes, pipe)
-		} else {
-			nonSelectedPipes = append(nonSelectedPipes, pipe)
-		}
-	}
+	selectedPipes, nonSelectedPipes := slices.Partition(pipes, func(pipe *Pipe) bool {
+		return highlight && equalHashes(pipe.fromSha, selectedCommitSha)
+	})
```

Comment:
Interesting case here: I opted to preallocate one of the two slices in the original code, keeping performance in mind. But this decision wasn't actually informed by any benchmarking, and I find the new version much more readable.